src/daemon/device_mouse.c

```diff
@@ -75,6 +75,25 @@ int setactive_mouse(usbdevice* kb, int active){
     return 0;
 }
 
+// FIXME: every time the battery is read, the mouse doesn't accept input, which makes using it quite uncomfortable to use
+void* read_battery(void* context){
+    usbdevice* kb = context;
+    uchar msg[MSG_SIZE] = { CMD_GET, FIELD_BATTERY };
+
+    // Loop while device isn't being closed
+    while(kb->read_battery) {
+        uchar in[MSG_SIZE] = {};
+        if(!usbrecv(kb, msg, in)){
```

Comment:
This is going to burn CPU cycles if `usbrecv` fails for some reason. Perhaps insert a `sleep(10)` here too?

Reply:
Fixed in new commit